backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,342 @@
+                                  boolean isNewInstance,
+                                  BacktraceBase staleBreadcrumbLogSender) {
+        if (isNewInstance == true) {
+            handleExistingBreadcrumbLogFile(context, staleBreadcrumbLogSender);
+        }
+        if (backtraceBreadcrumbsLogger == null) {
+            try {
+                backtraceBreadcrumbsLogger = new BacktraceBreadcrumbsLogger(getBreadcrumbLogPath(this.context), this.maxBreadcrumbLogSizeBytes);
+            } catch (Exception ex) {
+                BacktraceLogger.e(LOG_TAG, "Could not start the Breadcrumb logger due to: " + ex.getMessage());
+                return false;
+            }
+        }
+
+        registerAutomaticBreadcrumbReceivers(enabledBreadcrumbTypes);
+
+        this.enabledBreadcrumbTypes = enabledBreadcrumbTypes;
+
+        // We should log all breadcrumb configuration changes in the breadcrumbs
+        addConfigurationBreadcrumb();
+        return true;
+    }
+
+    public boolean enableBreadcrumbs(boolean isNewInstance, BacktraceBase staleBreadcrumbLogSender) {
+        final Set<BacktraceBreadcrumbType> enabledBreadcrumbTypes = new HashSet<BacktraceBreadcrumbType>(){{
+            add(BacktraceBreadcrumbType.CONFIGURATION);
+            add(BacktraceBreadcrumbType.HTTP);
+            add(BacktraceBreadcrumbType.LOG);
+            add(BacktraceBreadcrumbType.MANUAL);
+            add(BacktraceBreadcrumbType.NAVIGATION);
+            add(BacktraceBreadcrumbType.SYSTEM);
+            add(BacktraceBreadcrumbType.USER);
+        }};
+
+        return enableBreadcrumbs(enabledBreadcrumbTypes, isNewInstance, staleBreadcrumbLogSender);
+    }
+
+    private void unregisterAutomaticBreadcrumbReceivers() {
+        this.context.unregisterReceiver(backtraceBroadcastReceiver);
+        backtraceBroadcastReceiver = null;
+
+        this.context.unregisterComponentCallbacks(backtraceComponentListener);
+        backtraceComponentListener = null;
+
+        if (context instanceof Application) {
+            ((Application) context).unregisterActivityLifecycleCallbacks(backtraceActivityLifecycleListener);
+        }
+    }
+
+    public void disableBreadcrumbs() {
+        if (this.isBreadcrumbsEnabled()) {
+            unregisterAutomaticBreadcrumbReceivers();
+        }
+
+        // We should log all breadcrumb configuration changes in the breadcrumbs
+        enabledBreadcrumbTypes = null;
+        addConfigurationBreadcrumb();
+    }
+
+    public boolean clearBreadcrumbs() {
+        boolean success = backtraceBreadcrumbsLogger.clear();
+        // Make sure the configuration is always known
+        addConfigurationBreadcrumb();
+        return success;
+    }
+
+    public String getBreadcrumbLogDirectory() {
+        return context.getFilesDir().getAbsolutePath() + "/" + BacktraceBreadcrumbs.breadcrumbLogDirectory;
+    }
+
+    public static String getBreadcrumbLogPath(@NonNull Context context) {
```

Comment:
I will make `getBreadcrumgLogDirectory` static too. The reason I want this to be static is because we need to provide the attachment path to Crashpad when we initialize, and we initialize Crashpad before we might have a `BacktraceBreadcrumbs` instance

Reply:
Outdated.